aws/scripts/py/cache_invalidation.py

```
@@ -0,0 +1,302 @@
+            raise CloudFrontInvalidationError("Failed to create CloudFront client") from err
+
+    def _is_app_distribution(self, origin_domains: List[str]) -> bool:
+        """Check if this is an app distribution (should be skipped)"""
+        return any("app." in domain for domain in origin_domains)
+
+    def _is_staging_distribution(self, distribution: Dict[str, Any], origin_domains: List[str]) -> bool:
+        """Check if this is a staging distribution"""
+        # Check staging flag
+        if distribution.get("IsStagingDistribution", False):
+            return True
+        
```

Flake8 (7.2.0)

[warning] blank line contains whitespace

(W293)